lib/model/CIndividualModel.cc

```diff
@@ -339,6 +339,21 @@ void CIndividualModel::doPersistModelsState(core::CStatePersistInserter& inserte
     }
 }
 
+bool CIndividualModel::shouldPersistModel() const {
+    // If a feature model's memory usage is zero, it indicates that the model
+    // has been pruned and replaced with a stub. If all feature models are
+    // such stubs then we choose to not persist the model
+    // in order to reclaim memory after a persist/restore cycle.
+    for (const auto& feature : m_FeatureModels) {
+        for (const auto& model : feature.s_Models) {
```

Comment:
It feels a bit fragile to check memory usage here. I think it would make the code self documenting if you introduced a method `maths::CModel::shouldPersist` then this could be:
```
if (std::any_of(feature.s_Models.begin(), feature.s_Models.end(), [](const auto& model) { return model.shouldPersist(); }) {
  return true;
}
```

Reply:
I would also be inclined to move this inner loop into a method `SFeatureModels::shouldPersist()` which can then be shared by all implementations. With this, the implementation of this method just becomes:
```
  return std::any_of(m_FeatureModels.begin(), m_FeatureModels.end(), [](const auto& model) { return model.shouldPersist()});
```

Exactly the same check can be applied in `CMetricPopulationModel` and `CEventRatePopulationModel`. So for example:
```
bool CMetricPopulationModel::shouldPersist() const {
  return std::any_of(m_FeatureModels.begin(), m_FeatureModels.end(), [](const auto& model) { return model.shoulPersist(); });
}
```